Sources/MongoSwift/BSON/Document.swift

```diff
@@ -60,6 +69,7 @@ public struct Document: ExpressibleByDictionaryLiteral, ExpressibleByArrayLitera
      */
     internal init(fromPointer pointer: UnsafePointer<bson_t>) {
         self.storage = DocumentStorage(fromPointer: pointer)
+        self.count = Int(bson_count_keys(self.storage.pointer))
```

Comment:
I count (ha ha) 5 places now where we do `Int(bson_count_keys(some pointer))`. would it make sense to add an internal computed property on `Document` for this? idk what to name, `countFromData` or something maybe

Reply:
Is the idea that the computed property would always recompute? Document already has a `count` property, so would it make more sense for the always-computed property to go on `DocumentStorage` instead?

Reply:
A computed property is recomputed whenever it is accessed, I believe.

I don't think it would make too big of a difference either way, since we'll be making it `private` or `internal` with our two choices.

Reply:
yes, basically this would just be a wrapper for calling `bson_count_keys`. but it's more swifty to make it a computed property instead of a function. 

it could go either place, but thus far we have kept all methods that actually do things with the `bson_t` inside of `Document`, and left `DocumentStorage` as a class that does nothing but handle initialization and cleanup. I think that simplicity is kind of nice.

if we can come up with some sort of established rule for when something is a `DocumentStorage` method vs a `Document` method, I'm not opposed to moving some of our internal methods over there.

Reply:
Ah, I just remembered _why_ I didn't make this a computed property to begin with...

All the uses of `Int(bson_count_keys(self.storage.pointer))` is for setting `self.count`..., but if it is made into a computed property then we'd have to do something like `self.count = self.bsonCount` but this won't work because we are in an initializer before we have initialized all of the variables.

I think going to `DocumentStorage` would make this work, or we can make it a private function that takes a pointer.